bin/metrics-aggregate.rb

```diff
@@ -95,27 +94,27 @@ def acquire_checks
 
   def get_aggregate(check)
     uri = "/aggregates/#{check}"
-    issued = api_request(uri + "?age=#{config[:age]}")
-    unless issued.empty?
-      issued_sorted = issued.sort
-      time = issued_sorted.pop
-      unless time.nil?
-        uri += "/#{time}"
-        [time, api_request(uri)]
-      else
-        warning "No aggregates older than #{config[:age]} seconds"
-      end
-    else
+    issued = api_request(uri + "?max_age=#{config[:age]}")
+    if issued.empty?
       warning "No aggregates for #{check}"
+    else
+      issued
     end
   end
 
   def run
+    timestamp = Time.now.to_i
     acquire_checks.each do |check|
-      timestamp, aggregate = get_aggregate(check[:check])
-      puts "#{check[:check]} aggregates: #{aggregate}" if config[:debug]
+      aggregate = get_aggregate(check['name'])
+      puts "#{check['name']} aggregates: #{aggregate}" if config[:debug]
       aggregate.each do |result, count|
-        output "#{config[:scheme]}.#{check[:check]}.#{result}", count, timestamp
+        if count.is_a?(Hash)
```

Comment:
@majormoses the only hack is here so that the new endpoint doesnt screw up the existing datapoints

Reply:
can you add a comment on the line above as why it's needed so that we can remove it at some point in the future.

Reply:
basically something like:
```
# in 0.23 they changed the api results for aggregates this helps maintain backwards compatibility
if count.is_a?(Hash)
```

Reply:
Have not heard back from you on this, I am gonna add this comment on release.